graph_adventure/adv.py

```diff
@@ -161,11 +161,15 @@ def dft(world, startingRoom):
             if next_room not in visited:
                 visited[next_room] = direction
                 new_path = path + [next_room]
-                next_directions.append(direction)
 
-                stack.append(new_path)
+                next_paths.append(new_path)
 
-        if not len(next_directions) and len(stack):
+        if len(next_paths):
```

Comment:
I like the fact you included randomness in here into your solution Max!, great demonstration of what you learned from these 2 weeks.